### PR TITLE
[Test] Optimize Quick Draw tests

### PR DESCRIPTION
**Pull request by OrangeRed**
## What are the changes?
Optimize the quick draw tests so they don't take so long

## Why am I doing these changes?
Quick draw tests are timing out when run in CI

## What did change?
Stop the phase interceptor right before the faint phase before checking if the ability was applied / the pokemon have fainted

## How to test the changes?
`pnpm test:silent`

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)

### CODE REVIEW

File: src/test/abilities/quick_draw.test.ts [hunk 1]
```diff
@@ -28,90 +26,71 @@ describe("Abilities - Quick Draw", () => {
...
-      allAbilities[Abilities.QUICK_DRAW].getAttrs(BypassSpeedChanceAbAttr)[0],
-      "chance","get"
-    ).mockReturnValue(100);
+    vi.spyOn(Overrides, "SINGLE_BATTLE_OVERRIDE", "get").mockReturnValue(true);
+    vi.spyOn(Overrides, "ABILITY_OVERRIDE", "get").mockReturnValue(Abilities.QUICK_DRAW);
+    vi.spyOn(Overrides, "MOVESET_OVERRIDE", "get").mockReturnValue([Moves.TACKLE, Moves.TAIL_WHIP]);
+    vi.spyOn(Overrides, "OPP_SPECIES_OVERRIDE", "get").mockReturnValue(Species.RATTATA);
+    vi.spyOn(Overrides, "OPP_MOVESET_OVERRIDE", "get").mockReturnValue([Moves.TACKLE, Moves.TACKLE, Moves.TACKLE, Moves.TACKLE]);
+
+    vi.spyOn(allAbilities[Abilities.QUICK_DRAW].getAttrs(BypassSpeedChanceAbAttr)[0], "chance", "get").mockReturnValue(100);
```
Comment (flx-sta) [inline src/test/abilities/quick_draw.test.ts:35]: formatting noise? how come

**Review by Tempo-anon — DISMISSED** [review]

File: src/test/abilities/quick_draw.test.ts [hunk 2]
```diff
@@ -28,90 +26,71 @@ describe("Abilities - Quick Draw", () => {
...
   });
 
-  it("makes pokemon going first in its priority bracket", async() => {
+  test("makes pokemon going first in its priority bracket", async () => {
```
Comment (flx-sta) [inline src/test/abilities/quick_draw.test.ts:38]: whats the gain from using `test` instead of `it`?

**Review by flx-sta — APPROVED** [review]
lgtm 🚀 

just some minor open questions

File: src/test/abilities/quick_draw.test.ts (hunk 2 above)
Comment (OrangeRed) [reply src/test/abilities/quick_draw.test.ts:38]: I like the word `test` :nosepass:

Comment (flx-sta) [reply src/test/abilities/quick_draw.test.ts:38]: thats good for you 👀 
But every other test uses `it` and I think we should keep `it` that way :D

Comment (OrangeRed) [reply src/test/abilities/quick_draw.test.ts:38]: all the tests that I've written use the word test though?

File: src/test/abilities/quick_draw.test.ts (hunk 1 above)
Comment (OrangeRed) [reply src/test/abilities/quick_draw.test.ts]: prettier just doing prettier things

**Review by Tempo-anon — APPROVED** [review]

File: src/test/abilities/quick_draw.test.ts (hunk 2 above)
Comment (flx-sta) [reply src/test/abilities/quick_draw.test.ts:38]: you sneaky.
Well I guess it's survival of whatever stays then

**Pull request merged**